src/main/java/yoda/Ui.java

```diff
@@ -0,0 +1,78 @@
+package yoda;
+
+/**
+ * Encapsulates the user interface for the chatbot.
+ * @author Tonishka Singh
+ */
+public class Ui {
+    public Ui() {}
```

Comment:
This is similar to the case with the constructor for Parser. I think you can just delete it to simplify things and avoid violations.